fv3core/stencils/updatedzc.py

```diff
@@ -1,214 +1,72 @@
-#         enddo
-#      enddo
-
-#      if (grid_type < 3) call fill_4corners(gz2, 1, bd, npx, npy, sw_corner, se_corner, ne_corner, nw_corner) # noqa: E501
-
-#      do j=js1, je1
-#         do i=is1, ie2
-#            if( xfx(i,j) > 0.0 ) then
-#                fx(i,j) = gz2(i-1,j)
-#            else
-#                fx(i,j) = gz2(i  ,j)
-#            endif
-#            fx(i,j) = xfx(i,j)*fx(i,j)
-#         enddo
-#      enddo
-
-#      if (grid_type < 3) call fill_4corners(gz2, 2, bd, npx, npy, sw_corner, se_corner, ne_corner, nw_corner) # noqa: E501
-
-#      do j=js1,je2
-#         do i=is1,ie1
-#            if( yfx(i,j) > 0.0 ) then
-#                fy(i,j) = gz2(i,j-1)
-#            else
-#                fy(i,j) = gz2(i,j)
-#            endif
-#            fy(i,j) = yfx(i,j)*fy(i,j)
-#         enddo
-#      enddo
-
-
 @gtscript.function
 def xy_flux(gz_x, gz_y, xfx, yfx):
     fx = xfx * (gz_x[-1, 0, 0] if xfx > 0.0 else gz_x)
     fy = yfx * (gz_y[0, -1, 0] if yfx > 0.0 else gz_y)
     return fx, fy
 
 
-#      do j=js1, je1
-#         do i=is1,ie1
-#            gz(i,j,k) = (gz2(i,j)*area(i,j) +  fx(i,j)- fx(i+1,j)+ fy(i,j)- fy(i,j+1)) & # noqa: E501
-#                      / (         area(i,j) + xfx(i,j)-xfx(i+1,j)+yfx(i,j)-yfx(i,j+1)) # noqa: E501
-#         enddo
-#      enddo
-
-#   end do
-
-# ! Enforce monotonicity of height to prevent blowup
-#   do i=is1, ie1
-#      do j=js1, je1
-#         ws(i,j) = ( zs(i,j) - gz(i,j,km+1) ) * rdt
-#      enddo
-#   end do
-
-#   do k = km, 1, -1
-#      do j=js1, je1
-#         do i=is1, ie1
-#            gz(i,j,k) = max( gz(i,j,k), gz(i,j,k+1) + dz_min )
-#         enddo
-#      enddo
-#   enddo
-
-
 @gtstencil()
-def update_dz_c(
-    dp_ref: sd,
-    zs: sd,
-    area: sd,
-    ut: sd,
-    vt: sd,
-    gz: sd,
-    gz_x: sd,
-    gz_y: sd,
-    ws3: sd,
-    *,
-    dt: float,
+def update_dz_c_stencil(
+    area: FloatField,
+    dp_ref: FloatField,
+    z_surface: FloatField,
+    ut: FloatField,
+    vt: FloatField,
+    gz: FloatField,
+    ws: FloatField,
+    dt2: float,
 ):
+    """Update the model heights from the C-grid wind flux
+
+    After the model runs c_sw and advances the c-grid variables half a timestep,
+    the grid deforms with the flow in the Lagrangian coordinate system. This
+    module updates the model heights based on the flux, and the rate of change of
+    the lowest model height compared to the fixed surface height.
+
+    Args:
+         dp_ref: vertical delta in column reference pressure(in)
+         z_surface: surface height (m) (in)
+         ut: x-velocity on the C-grid, contravariantof the D-grid winds(in)
+         vt: y-velocity on the C-grid, contravariantof the D-grid winds(in)
+         gz: height of the model grid cells (m) (inout)
+         ws: difference in height between lowest model layer and the surface per C-grid
+             timestep(inout)
+         dt2: half a model timestep (for C-grid update) in seconds (in)
```

Comment:
I notice `dt -> dt2`, I'm a little concerned with this name being ambiguous/providing more context than this function needs. Out of context, in "half a model timestep for C-grid update", "model timestep for C-grid update" means half a dynamics timestep, so the phrase means 1/4th of the model timestep (because the model gets progressed forward half a timestep on each grid type).

Maybe that's actually what it means (1/4th of the dynamics timestep), I actually don't know.

When I say "providing more context than the function means", I mean the function is saying where you're calling it from (somewhere that has a timestep twice as long as this function does), but you're allowed to call it from a different context. Once we've learned/clarified what this function is doing, it could be called by a different model in someone's analysis script with a timestep that has nothing to do with FV3.

Would it make sense to provide this as just `dt` and say it's the timestep over which the dz update is taking place? dt gets used strictly to calculate vertical velocity, so this would be the most precise. You could add a clarification sentence along the lines of "Note within the FV3 dynamical core this is half a dynamics time step".